src/App.jsx

```diff
@@ -4,22 +4,47 @@ import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
 
+const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+
+const headers = {
+  Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+  'Content-Type': 'application/json',
+};
+
+
 const App = () => {
 
   const [todoList, setTodoList] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
+
+  const fetchData = async () => {
+    const options = {
+      method: "GET",
+      headers: headers,
+    };
+    try {
+      const response = await fetch(url, options);
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+      const data = await response.json();
+      const todos = data.records.map(record => {
+        const todo = {
+          id: record.id,
+          title: record.fields.title
+        };
+        return todo;
+      });
+      setTodoList(todos);
+    } catch (error) {
+      console.log(error);
```

Comment:
same here add a setIsLoading(false) here because at this point the data has finished loading

Reply:
why do I need to add `setIsLoading(false);` twice when I can add this line only once after the try-catch block?_

Reply:
you can add it before the `if (!response.ok) {` statement.

You should not add it after the trycatch block because if you have a loading UI or an error UI and you need to show them, you will both have an error state and a loading state.

so it is better to stop loading immediately you get your data or an error

Reply:
Got it. Thank you for the explanation!

Reply:
Done